linalg/sparsemat.cpp

```diff
@@ -1173,15 +1174,23 @@ void SparseMatrix::EliminateRowCol(int rc, const double sol, Vector &rhs,
       {
          if ((col = J[j]) == rc)
          {
-            if (d)
-            {
-               rhs(rc) = A[j] * sol;
-            }
-            else
-            {
-               A[j] = 1.0;
-               rhs(rc) = sol;
-            }
+           switch (dpolicy)
+           {
+           case DIAG_KEEP:
+             rhs(rc) = A[j] * sol;
+             break;
+           case DIAG_ONE:
+             A[j] = 1.0;
+             rhs(rc) = sol;
+             break;
+           case DIAG_ZERO:
+             A[j] = 0.;
+             rhs(rc) = sol;
```

Comment:
If we follow the same pattern as the other two cases this line should be rhs(rc)=0.0;.  However, I'm not sure how this might be used.  Depending on the use case it could also make sense to leave the right hand side unchanged.